libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -157,27 +187,85 @@ class FeatureFlagConfigurationSettingsConverter implements ConfigurationSettings
   Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<FeatureFlagValue>[] {
-    const settings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
+  ): SetConfigurationSettingParam<MsFeatureFlagValue>[] {
+    const settings = new Array<SetConfigurationSettingParam<MsFeatureFlagValue>>();
+    const featureFlags = new Array<MsFeatureFlagValue>();
 
-    for (const featureFlag in config) {
-      if (!this.validateFeatureName(featureFlag)) {
-        throw new ArgumentError(
-          `Feature flag ${featureFlag} contains invalid character,'%' and ':' are not allowed in feature name. Please provide valid feature name.`
-        );
+    if (this.foundMsFeatureManagement) {
+      const msFmSectionFeatureFlags = this.getMsFmSchemaFeatureFlags(config);
+
+      if (msFmSectionFeatureFlags) {
+        for (const featureFlag of msFmSectionFeatureFlags) {
+          if (!featureFlag.id) {
+            throw new ArgumentError(
+              "Feature flag without id is found, id is a required property."
+            );
+          }
+
+          if (!this.validateFeatureName(featureFlag.id)) {
+            throw new ArgumentError(
+              `Feature flag id ${featureFlag.id} contains invalid character,'%' and ':' are not allowed.`
+            );
+          }
+
+          const featureFlagDefinition = this.getFeatureFlagDefinitionFromMsFmSchema(featureFlag);
+
+          // Check if the featureFlag with the same id already exists
+          // Replace the existing flag with the later one, the later one always wins
+          const indexOfExistingFlag = featureFlags.findIndex(existingFeatureFlag => existingFeatureFlag.id === featureFlag.id);
+
+          if (indexOfExistingFlag !== -1) {
+            featureFlags[indexOfExistingFlag] = featureFlagDefinition;
+          } 
+          else {
+            featureFlags.push(featureFlagDefinition);
+          }
+        }
+      }
+    }
+
+    if (this.legacySchemaFeatureManagementKeyWord) {
+      const legacySchemaFeatureFlags = this.getLegacySchemaFeatureFlags(config);
+
+      if (legacySchemaFeatureFlags) {
+        for (const featureFlag in legacySchemaFeatureFlags) {
+          if (!this.validateFeatureName(featureFlag)) {
```

Comment:
To confirm on line 232 we are passing the feature flag name to `validateFeatureName` ?

Reply:
Yes in the legacy schema the the feature name is the key in the json schema